eng/native/configurecompiler.cmake

```diff
@@ -319,6 +319,7 @@ if (CLR_CMAKE_HOST_UNIX)
     endif()
   endif(CLR_CMAKE_HOST_OSX OR CLR_CMAKE_HOST_MACCATALYST)
 
+  add_compile_options(-Werror=implicit-function-declaration)
```

Comment:
src/libraries/Native is building as part of coreclr as well as libraries these days. `configurecompiler.cmake` is currently used by all subsets excluding libs for reasons, so this would need to be next to https://github.com/dotnet/runtime/blob/aff3362fd7743753b6e86a77d6f934e7a7185bc1/src/libraries/Native/Unix/System.Security.Cryptography.Native/CMakeLists.txt#L7 for it to make a difference for OpenSSL, when someone is {re}building the libs subset.

Reply:
(`configuretools.cmake` is the one shared by libs, but compiler flags aren't defined there)